src/fabric.ts

```diff
@@ -193,187 +220,127 @@ export class FabricListener implements ManagedResource {
         }
     }
 
-    private getMessageTimestamp(msg: FabricClient.BlockData): Date {
+    private getMessageTimestamp(msg: BlockData): Date {
         return new Date(get(msg, 'payload.header.channel_header.timestamp'));
     }
 
-    private processBlock = (channelName: string, initCheckpoint: number) => (
-        block: FabricClient.Block | FabricClient.FilteredBlock
-    ) => {
-        if (this.isFilteredBlock(block)) {
-            return this.processFilteredBlock(block);
-        }
-
-        const blockNumber = +block.header.number;
-        if (blockNumber <= initCheckpoint) {
-            debug(`Ignoring block number=%d on channel=%d since we already processed it`, blockNumber, channelName);
-            return;
-        }
-        debug('Processing block number=%d on channel=%s', blockNumber, channelName);
+    private processBlock: BlockListener = async (event) => {
+        debug('Got block number %d block data %s', event.blockNumber, JSON.stringify(event.blockData));
+        const block = event.blockData as any;
+        if (block && block.data) {
+            const channelName = String(this.getChannelId(block.data.data[0]));
+            const initCheckpoint = this.checkpoint.getChannelCheckpoint(channelName);
+            const blockNumber = Number(event.blockNumber);
+            if (blockNumber <= initCheckpoint) {
+                debug(`Ignoring block_number=%d on channel=%d since we already processed it`, blockNumber, channelName);
+                return;
+            }
 
-        for (const msg of block.data.data) {
-            if ('payload' in msg) {
-                this.parseMessageHeaderExtension(msg);
-                this.parseChaincodeSpecInput(msg);
+            debug('Processing block_number=%d on channel=%s', blockNumber);
```

Comment:
```suggestion
            debug('Processing block number=%d on channel=%s', blockNumber, channelName);
```

Could also consider doing this in `info` level